source/OptimaJet.DWKit.StarterApplication/Services/UserService.cs

```diff
@@ -11,14 +14,60 @@ namespace OptimaJet.DWKit.StarterApplication.Services
     public class UserService : EntityResourceService<User>
     {
         public UserRepository EntityRepository { get; }
-
+        public IJsonApiContext JsonApiContext { get; }
+        public IOrganizationContext OrganizationContext { get; }
+        public ICurrentUserContext CurrentUserContext { get; }
 
         public UserService(
             IJsonApiContext jsonApiContext,
+            IOrganizationContext organizationContext,
+            ICurrentUserContext currentUserContext,
             IEntityRepository<User> entityRepository,
             ILoggerFactory loggerFactory) : base(jsonApiContext, entityRepository, loggerFactory)
         {
             this.EntityRepository = (UserRepository)entityRepository;
+            JsonApiContext = jsonApiContext;
+            OrganizationContext = organizationContext;
+            CurrentUserContext = currentUserContext;
+        }
+
+        public override async Task<IEnumerable<User>> GetAsync()
+        {
+            if (OrganizationContext.SpecifiedOrganizationDoesNotExist)
+            {
+                return Enumerable.Empty<User>().AsQueryable();
+            }
+            else
+            {
+                var query = JsonApiContext.QuerySet;
+                if (query == null)
+                {
+                    query = new JsonApiDotNetCore.Internal.Query.QuerySet();
+                    JsonApiContext.QuerySet = query;
+                }
+                var value = OrganizationContext.HasOrganization ? OrganizationContext.OrganizationId.ToString() : "";
+                query.Filters.Add(new JsonApiDotNetCore.Internal.Query.FilterQuery("organization-header", value, "="));
+                return await base.GetAsync();
+            }
+        }
+        public override async Task<User> GetAsync(int id)
+        {
+            var currentUser = EntityRepository.GetByAuth0Id(CurrentUserContext.Auth0Id).Result;
+            if (currentUser.Id == id)
+            {
+                return await base.GetAsync(id);
+            }
+            var users = await GetAsync();
+            return users.SingleOrDefault(u => u.Id == id);
+        }
+        public override async Task<User> UpdateAsync(int id, User resource)
```

Comment:
this looks like it should almost by default behavior, yeah?

Reply:
As with an earlier one, no.  The base calls the repository getAsync(id) directly without calling the filters etc.  The most straightforward way to implement the required checks was to getAsync() getting all and then pulling the requested one out of the list if it is still there.  Chris did send an email to you when we implemented asking if you had any objections.